File: data_augmentation/code_cpp/translate_replicate.cpp
#include"opencv2/opencv.hpp"
#include<iostream>
using namespace cv;
using namespace std;

int main(int argc,char *argv[])
{
    Mat img=imread(argv[1]);
    Mat resized;
    Mat translated;
    unsigned pos;
    string filename(argv[1]);
    string subname;
    string cname;

    pos=filename.find(".jpg");
    subname=filename.substr(0,pos);
    
    
    //Resize it to 128x128
    resize(img,resized,Size(128,128));
     
    //Top left 120*120
    translated=resized(Rect(0,0,120,120));
    cname=subname+"tl.jpg";
    imwrite(cname,translated);
    
    //Top right 120*120
    translated=resized(Rect(8,0,120,120));
    cname=subname+"tr.jpg";
    imwrite(cname,translated);
    
    //Bottom Left 
    translated=resized(Rect(0,8,120,120));
    cname=subname+"bl.jpg";
    imwrite(cname,translated);

    //Bottom Right
    translated=resized(Rect(8,8,120,120));
    cname=subname+"br.jpg";
    imwrite(cname,translated);
    
    //Center 
    translated=img(Rect(4,4,120,120));
    cname=subname+"c.jpg";
    imwrite(cname,translated);
    
 

    
    //imshow("resized",resized);
    //imwrite("resized.jpg",resized);
    //waitKey(0);
    



}

File: data_augmentation/code_python/translate_replicate.py
from cv2 import *
import sys
def fn(img):
    
    #resizing it to 128*128
    resized=resize(img,((128,128)))
    namedWindow("translated") 
    #getting translated images
    for i in [0,4,8]:
        for j in [0,4,8]:
            if(i==4 or  j==4):
                if(i!=4 or j!=4):
                    continue
            translated=resized[i:i+120,j:j+120]
            imshow("translated",translated)
            waitKey(0)
    
        
  

if __name__=='__main__':
    print sys.argv[1]
    img=imread(sys.argv[1])
    
    fn(img)
